### PR TITLE
Caching + dev details

### PR DESCRIPTION
**Pull request by natebrennand**

### CODE REVIEW

File: integrations/persistence.go [hunk 1]
```diff
@@ -70,6 +88,12 @@ func (c Client) GetUser(idx Index, value string) (User, error) {
 
 // GetUserList returns all users.
 func (c Client) GetUserList() ([]User, error) {
+	// return cached users if they were refreshed in the last 10 minutes
+	if c.cache.lastUpdated.Add(cacheTTL).After(time.Now()) {
+		log.Println("Hitting cache")
```
Comment (xavi-) [inline integrations/persistence.go]: Any concern that this will spam the logs?

**Comment by xavi-** [comment]
lgtm.

Out curiosity, any thoughts on utilizing a cache for `GetUser`?

File: integrations/persistence.go (hunk 1 above)
Comment (natebrennand) [inline integrations/persistence.go]: oops, this was debugging. will remove

**Comment by natebrennand** [comment]
I considered it but moved away because the interface was going to be nastier for handling purging. I guess I could purge everything every N minutes.
Do you think it is worthwhile to cache it all?

**Comment by xavi-** [comment]
Maybe... I'm not sure what who's who usage patterns will look like, but I could imagine traffic to `/alias/aws/:username` (and friends) being very bursty.  For example, a fairly reasonable implementation of aws email notifier might make a unique `/alias/...` call for each aws user name it finds, which could push who's who over it's dynamo rate limit.  Not sure how realistic that is, though.

**Comment by natebrennand** [comment]
Throughput is watched in a 5 minute window, 300 seconds. With the current throughput allocation of 2 read/sec, we could handle a burst of 600 requests.

I think it's more reasonable to up the throughput if it becomes a big issue rather than complicate the code more. DynamoDB pricing is [very cheap](http://aws.amazon.com/dynamodb/pricing/) (especially for reads).

**Comment by natebrennand** [comment]
Should we just bump every index to a throughput of 5 and then never worry about it?
The caching for `/list` is only there because it burns through the throughput much quicker.

**Comment by xavi-** [comment]
Nah, if can handle burst of 600, should be fine.

**Comment by xavi-** [comment]
lgtm.

**Pull request merged**